core/src/main/scala/kafka/server/checkpoints/CheckpointFile.scala

```diff
@@ -60,17 +59,27 @@ class CheckpointReadBuffer[T](location: String,
               case Some(e) =>
                 entries += e
                 line = reader.readLine()
-              case _ => throw malformedLineException(line)
+              case _ => return maybeThrowException(malformedLineException(line))
             }
           }
           if (entries.size != expectedSize)
-            throw new IOException(s"Expected $expectedSize entries in checkpoint file ($location), but found only ${entries.size}")
+            return maybeThrowException(new IOException(s"Expected $expectedSize entries in checkpoint file ($location), but found only ${entries.size}"))
           entries
         case _ =>
-          throw new IOException(s"Unrecognized version of the checkpoint file ($location): " + version)
+          maybeThrowException(new IOException(s"Unrecognized version of the checkpoint file ($location): " + line))
       }
     } catch {
-      case _: NumberFormatException => throw malformedLineException(line)
+      case _: NumberFormatException => maybeThrowException(malformedLineException(line))
+    }
+  }
+
+  private def maybeThrowException(e: Exception): Seq[T] = {
```

Comment:
nit: The `maybeThrow` sounds a bit vague to me.  How about `dropCorruptFileOrThrow`

Reply:
Thanks for the suggestion. I've updated the method name.